shm/Ship.cpp

```diff
@@ -41,20 +42,37 @@ Ship& Ship::operator+=(const size_t crew) {
 }
 
 void Ship::load(const std::shared_ptr<Cargo>& cargo) {
-    cargo_.push_back(cargo);
+    auto it = std::find_if(begin(cargo_), end(cargo_),
+                           [&cargo](const auto& ptr) {
+                               return *cargo == *ptr;
+                           });
+
+    if (it != end(cargo_)) {
+        *(*it) += cargo->getAmount();
+    } else {
+        cargo_.push_back(cargo);
+    }
+}
+
+void Ship::unload(const std::shared_ptr<Cargo>& cargo, size_t amount) {
+    if (amount == cargo->getAmount()) {
+        cargo_.erase(std::remove(begin(cargo_), end(cargo_), cargo), cargo_.end());
+    } else {
+        *cargo -= amount;
+    }
 }
 
-void Ship::unload(Cargo* cargo) {
-    if (!cargo->getAmount()) {
-        std::remove_if(cargo_.begin(), cargo_.end(), [cargo](const auto& ptr) {
-            return ptr.get() == cargo;
-        });
+void Ship::print() const {
```

Comment:
I think it could be done with overloaded << operator

Reply:
yeah but someone else has << operator in store, so I quess he could do this aswell

Reply:
I've done operator<< for Ship class on my branch "crazyShopping", but it was done with `basePrice`

```cpp
std::ostream& operator<<(std::ostream& out, const Ship& ship) {
    std::string horizontalSeparator(61, '=');
    std::string headerSeparator(22, '=');
    int i = 0;
    out << "\n" << headerSeparator << " SHIP'S  STOCK " << headerSeparator
              << "\n"
              << "|| NAME"
              << std::setw(27) << "| QTY "
              << std::setw(5) << "| BASE PRICE "
              << std::setw(3) << " ||\n"
              << horizontalSeparator << "\n";

    for (const CargoPtr& el : ship.cargo_) {
        out << "||"
                  << std::setw(2) << ++i << ". "
                  << std::setw(21) << std::left << el->getName() << " | "
                  << std::setw(3) << std::right << el->getAmount() << " | "
                  << std::setw(10) << std::right << el->getBasePrice() << " ||\n";
    }
    out << horizontalSeparator << "\n";
    
    return out;
}
```